proof/refine/AARCH64/ArchStateRelation.thy

```diff
@@ -6,53 +6,30 @@
  *)
 
 (*
-   The refinement relation between abstract and concrete states
+   The refinement relation between abstract and concrete states - architecture-specific definitions.
 *)
 
-theory StateRelation
+theory ArchStateRelation
 imports ArchInvariantUpdates_H
 begin
 
-context begin interpretation Arch .
-
-definition cte_map :: "cslot_ptr \<Rightarrow> machine_word" where
-  "cte_map \<equiv> \<lambda>(oref, cref). oref + (of_bl cref << cte_level_bits)"
-
-lemmas cte_map_def' = cte_map_def[simplified cte_level_bits_def shiftl_t2n mult_ac, simplified]
+(* FIXME arch-split: these are generic, but due to many definitions, the Arch theory goes before
+   the generic one, so there's nowhere sensible to move to *)
+type_synonym obj_relation_cut = "Structures_A.kernel_object \<Rightarrow> Structures_H.kernel_object \<Rightarrow> bool"
+type_synonym obj_relation_cuts = "(machine_word \<times> obj_relation_cut) set"
```

Comment:
A small `_pre` theory?

Reply:
I tried to avoid creating yet another pre theory given it only needed a few lines. I'm mindful that pretty much every theory is going to become two, and Invariants_H already became 4 theories, so to have this one be 4 as well... if you think that's an improvement, I'll do it.

Reply:
this is done now